contracts/upgradeable_contracts/multi_amb_erc20_to_erc677/HomeMultiAMBErc20ToErc677.sol

```diff
@@ -249,10 +249,12 @@ contract HomeMultiAMBErc20ToErc677 is BasicMultiAMBErc20ToErc677, HomeFeeManager
         if (!lock()) {
             bytes32 _messageId = messageId();
             uint256 valueToBridge = _value;
-            uint256 fee = _distributeFee(HOME_TO_FOREIGN_FEE, _token, valueToBridge);
-            if (fee > 0) {
-                emit FeeDistributed(fee, _token, _messageId);
-                valueToBridge = valueToBridge.sub(fee);
+            if (!isRewardAddress(_from)) {
```

Comment:
Imagine that we have a dozen of the fee recipients so I think that others receive their fraction of fee from this withdrawal operation. So, I would say that `_distributeFee` must accept a parameter with the the current sender and if the sender is the fee recipient don't transfer tokens to it. The resulting fee will be `calculateFee(_feeType, _token, _value)` minus `feePerAccount` (`diff` also should be taken into account)

Reply:
I think this won't solve the general problem. Imagine a situation where we have 2 reward addresses, 10% fee, first reward account owns 100 tokens. 
If first reward account withdraws 100 tokens, second account will receive 10% / 2 = 5% fee, that is 5 tokens.
If now second account will try to withdraw his 5 tokens, first account will receive 0.25 tokens. So there won't be a finite way to withdraw all tokens from the home side.

Reply:
Understood. So, let's add the comment to this line of the code in order to not forget this clarification.